File: 3994-find-the-least-frequent-digit/find-the-least-frequent-digit.java
class Solution {
    public int getLeastFrequentDigit(int n) {
        int[] freq = new int[10];
        if (n == 0) {
            freq[0] = 1;
        } else {
            int x = n;
            while (x > 0) {
                int d = x % 10;
                freq[d]++;
                x /= 10;
            }
        }
        int minFreq = Integer.MAX_VALUE;
        for (int d = 0; d <= 9; d++) {
            if (freq[d] > 0 && freq[d] < minFreq) {
                minFreq = freq[d];
            }
        }
        for (int d = 0; d <= 9; d++) {
            if (freq[d] == minFreq) return d;
        }
        return 0;
    }
}

File: 3994-find-the-least-frequent-digit/find-the-least-frequent-digit.py
class Solution:
    def getLeastFrequentDigit(self, n: int) -> int:
        freq = dict()
        s = str(n)
        for i in s:
            freq[i] = s.count(i)
        ones = [d for d, c in freq.items() if c == 1]
        if len(ones) > 2:
            return min(int(d) for d in ones)
        min_freq = min(freq.values())
        candidates = [int(d) for d, c in freq.items() if c == min_freq]
        return min(candidates)